pczt/src/orchard.rs

```diff
@@ -0,0 +1,535 @@
+use std::collections::BTreeMap;
+
+#[cfg(feature = "orchard")]
+use ff::PrimeField;
+use getset::Getters;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+use crate::{
+    common::Zip32Derivation,
+    roles::combiner::{merge_map, merge_optional},
+};
+
+/// PCZT fields that are specific to producing the transaction's Orchard bundle (if any).
+#[derive(Clone, Debug, Serialize, Deserialize, Getters)]
+pub struct Bundle {
+    /// The Orchard actions in this bundle.
+    ///
+    /// Entries are added by the Constructor, and modified by an Updater, IO Finalizer,
+    /// Signer, Combiner, or Spend Finalizer.
+    #[getset(get = "pub")]
+    pub(crate) actions: Vec<Action>,
```

Comment:
This looks from `get = "pub"` as though it is only providing read access, but actually it generates an `actions_mut` accessor (and similarly for `{spends,outputs}_mut` on the Sapling bundle and  `{inputs,outputs}_mut` on the transparent bundle). I dislike the implicitness of `getset` tbh. I'm not saying we should get rid of it at this stage, but I'd have pushed back on it earlier if I'd realized it was implicitly generating mutable accessors. This makes it very difficult to determine whether invariants are preserved. In particular, the modifiability flags *don't* seem to be enforced. If there were explicit methods to modify these vectors, they could be.

Reply:
It doesn't implicitly generate mutable accessors; you have to explicitly request them with both a `derive(MutGetters)` and altering this to include `get_mut = "pub"`.

Reply:
I think I was looking at a previous state; there was a force-push that removed `get_mut` iirc. But the main point is that there is mutation of the `{spends,outputs,actions}` vectors that is a) hard to audit, and b) doesn't check the flags, which violates [a "must" in BIP 370](https://github.com/bitcoin/bips/blob/master/bip-0370.mediawiki#constructor) if it's this crate's responsibility.

Reply:
This thread was basically just an expression of my frustration at the incomplete `getset` documentation.